sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_shared/models.py

```diff
@@ -95,19 +120,21 @@ class PhoneNumberIdentifier(object):
     """
     kind = CommunicationIdentifierKind.PHONE_NUMBER
 
-    def __init__(self, value, **kwargs):
-        # type: (str, Any) -> None
+    def __init__(self, value: str, **kwargs: Any) -> None:
         self.raw_id = kwargs.get('raw_id')
         self.properties = PhoneNumberProperties(value=value)
         if self.raw_id is None:
             self.raw_id = _phone_number_raw_id(self)
 
+    def __eq__(self, other):
+        return self.raw_id == _phone_number_raw_id(other)
```

Comment:
Every `__eq__` method needs the same error handling you added above.
```suggestion
        try:
            return self.raw_id == _phone_number_raw_id(other)
        except Exception:
            return False
```

Reply:
Thanks for the comment! Updated!